pallets/common/src/traits/asset.rs

```diff
@@ -64,4 +68,15 @@ pub trait Trait<V, U> {
         ticker: &Ticker,
         value: V,
     ) -> DispatchResultWithPostInfo;
+    fn base_create_asset(
```

Comment:
Are there ever going to be more than one implementation of this?
Otherwise, are you doing this to break a circular dependency?
If we can avoid unnecessary abstractions generics, then that's good.

Reply:
Yes, we are doing this to avoid a circular dependency, these traits are mostly used to call a functionality of a pallet from another pallet, in this case we are keeping confidential asset's logic separate from regular asset logic, but because of some business requirements we still need to use a single storage (placed in the asset pallet) to make sure the asset wasn't registered before, etc.

Reply:
Hmm; but neither the confidential crate nor the asset crate refer to each other at the moment, so the confidential one could ostensibly add the asset crate as a dependency... I guess I'm not seeing where the circular dependency would arise from. Can you elaborate on that?

Reply:
to create a confidential asset I need to access the storage in both `asset` and `confidential-asset` pallets, confidential asset pallet already depends on the asset pallet, if I don't break out the logic like I did, and I put `create_confidential_asset()` in the asset pallet, I'll have to make `asset` depend on the confidential pallet, and that's a circular dependency.